### PR TITLE
Game test on master

### PR DESCRIPTION
**Pull request by NielsOuvrard**

### CODE REVIEW

File: src/entities/enemy.py [hunk 1]
```diff
@@ -5,11 +5,108 @@
...
+
+        if not self.target_in_range(5):
+            self.change_state(EnemyState.WALKING)
+
+    def update_dead(self):
+        """
+        Behavior for the DEAD state.
+        """
+        self.current_animation = 'die'
+        self.kill
```
Comment (NielsOuvrard) [inline src/entities/enemy.py]: self.kill ?

This line does not do anything

File: src/entities/enemy.py [hunk 2]
```diff
@@ -5,11 +5,108 @@
...
+        self.target: Player = None
+        self.state = EnemyState.IDLE  
+        self.health = 1
+
+    def change_state(self, new_state: EnemyState):
+        """
+        Change the current state of the enemy.
+        """
+        if self.state != new_state:  # Avoid redundant state changes
+            print(f"Transitioning from {self.state} to {new_state}")
```
Comment (NielsOuvrard) [inline src/entities/enemy.py]: ok to put print for debug purposes, but you should remove them when you want to put your commits on `master`

File: src/entities/enemy.py [hunk 3]
```diff
@@ -5,11 +5,108 @@
...
+        self.change_animation('attacking')
+        self.velocity = pg.Vector2(0, 0)  
+
+        if self.target_in_range(5):
+            player.take_damage(1)
+
+        if not self.target_in_range(5):
+            self.change_state(EnemyState.WALKING)
+
+    def update_dead(self):
```
Comment (NielsOuvrard) [inline src/entities/enemy.py]: put all of your update_*** function inside the update function directly to improve readability, and keep a structured class 👍

File: src/world/level.py [hunk 4]
```diff
@@ -63,7 +63,7 @@ class LevelHandler:
     def __init__(self) -> None:
         self.levels: list[Level] = []
         self.load_levels(TOML_FILE)
-        self.current_level: Level = self.levels[0]
+        self.current_level: Level = self.levels[1]
```
Comment (NielsOuvrard) [inline src/world/level.py]: should not be push, some commits after we change it for `level_selected`

File: src/ui/menu.py [hunk 5]
```diff
@@ -77,6 +78,14 @@ def draw(self, screen: pg.Surface) -> None:
     def handle_event(self, event: pg.event.Event) -> None:
         if event.type == pg.QUIT:
             self.next_state = MainState.QUIT
+        if self.level_handler.game_over:
+            self.next_state = MainState.MAIN_MENU
+        elif self.level_handler.current_level.finished:
+            time.sleep(2)
+            # print("buenas")
```
Comment (NielsOuvrard) [inline src/ui/menu.py:90]: you've forgot to remove this

File: src/world/level.py [hunk 6]
```diff
@@ -3,7 +3,7 @@
  # @ Create Time: 2024-11-12 19:40:33
  # @ Description:
  '''
-
+import time
```
Comment (NielsOuvrard) [inline src/world/level.py]: unused

File: src/world/level.py [hunk 7]
```diff
@@ -14,6 +14,8 @@
 from src.config import TILE_SIZE
 from src.core import Graph, Node, dijkstra
 
+from src.game_states import MainState
```
Comment (NielsOuvrard) [inline src/world/level.py:17]: unused

File: src/world/level.py [hunk 8]
```diff
@@ -62,8 +65,10 @@ class LevelHandler:
     """
     def __init__(self) -> None:
         self.levels: list[Level] = []
+        self.level_number = 0
+        self.game_over = False
```
Comment (NielsOuvrard) [inline src/world/level.py:70]: `game_over` name is ambiguous

why not `game_ended`

File: src/ui/menu.py [hunk 9]
```diff
@@ -77,6 +78,14 @@ def draw(self, screen: pg.Surface) -> None:
...
+            self.level_handler.next_level()
+            self.player.position = pg.Vector2(self.level_handler.current_level.start_position[0] * TILE_SIZE, self.level_handler.current_level.start_position[1] *TILE_SIZE)
+            self.enemies: list[Enemy] = [Enemy(pos) for pos in self.level_handler.current_level.enemies]
```
Comment (NielsOuvrard) [inline src/ui/menu.py]: I agree for the code, but it should not be here
Better a "reset" or "load_new_level" function in the class directly

File: assets/enemies.png
Comment (NielsOuvrard) [inline assets/enemies.png:1]: what have you changed in the picture ?

File: src/entities/enemy.py [hunk 10]
```diff
@@ -24,8 +25,8 @@ class Enemy(Entity):
     """
     def __init__(self, position: tuple[int, int]) -> None:
         super().__init__("assets/enemies.png", "assets/enemies.toml", position)
-        self.target: Player = None
-        self.state = EnemyState.IDLE  
+        self.target = player
```
Comment (NielsOuvrard) [inline src/entities/enemy.py:29]: if you add `player` here, update the`__init__` parameters in the same commit

File: src/entities/collisions.py [hunk 11]
```diff
@@ -123,4 +123,15 @@ def handle_collision(level: Level, position: pg.Vector2, image: pg.Surface, velo
             tile.color = (0, 0, 255)
     if sides_colliding != set():
         snap_position(sides_colliding, objects_colliding, position, image, velocity)
+
+    if is_player == "Player":
```
Comment (NielsOuvrard) [inline src/entities/collisions.py]: the variable `is_player` is ambiguous here, if you use it as a boolean, change the type
If `is_player` can be something else, rename it

File: src/world/level.py (hunk 8 above)
Comment (DJ2513) [reply src/world/level.py:70]: Idk, it has been always like that in all games, so it sounded correctly

File: assets/enemies.png
Comment (DJ2513) [reply assets/enemies.png:1]: I tried resolving the warning I told you the other day, but it still the same image

File: src/entities/enemy.py (hunk 10 above)
Comment (DJ2513) [reply src/entities/enemy.py:29]: Okey

File: src/entities/collisions.py (hunk 11 above)
Comment (DJ2513) [reply src/entities/collisions.py]: Yeah I took that out

File: src/world/level.py (hunk 8 above)
Comment (NielsOuvrard) [reply src/world/level.py:70]: `game_over` refers to loose
here the variable refers to `game finished`

It's ambiguous

File: assets/enemies.png
Comment (NielsOuvrard) [reply assets/enemies.png:1]: ok ok

If there is no modifications in the image, just remove it from your commit 👍

**Pull request merged**